docs/architecture/adr-038-state-listening.md

```diff
@@ -222,9 +222,12 @@ type ABCIListener interface {
 	ListenEndBlock(ctx types.Context, req abci.RequestEndBlock, res abci.ResponseEndBlock) error
 	// ListenDeliverTx updates the steaming service with the latest DeliverTx messages
 	ListenDeliverTx(ctx types.Context, req abci.RequestDeliverTx, res abci.ResponseDeliverTx) error
-	// ListenSuccess returns a chan that is used to acknowledge successful receipt of messages by the external service
-	// after some configurable delay, `false` is sent to this channel from the service to signify failure of receipt
-	ListenSuccess() <-chan bool
+	// HaltAppOnDeliveryError whether or not to halt the application when delivery of massages fails
+	// in ListenBeginBlock, ListenEndBlock, ListenDeliverTx. Setting this to `false` will give fire-and-forget semantics.
+	// When `true`, the app will gracefully halt and stop the running node. Uncommitted blocks will
+	// be replayed to all listeners when the node restarts and all successful listeners that received data
+	// prior to the halt will receive duplicate data.
+	HaltAppOnDeliveryError() bool
```

Comment:
"Setting this to `false`" is not possible, the method accepts no parameters and returns a `bool`?

Reply:
Ah! It just returns a const parameter. The wording makes it seem like it allows the caller to change that setting. So 

> HaltAppOnDeliveryError returns true if the application has been configured to . . .
> and false if it will . . . 
>
> This behavior is controlled by a corresponding app config setting.

Reply:
updated.